### DISCUSSION TITLE
Trouble using external binary files

### DISCUSSION THREAD

**Discussion by SebastianRuffert** [epochpic/epoch#409] [Q&A]
Hi,
I am trying to use raw binary files to specify the electron density. I create those binary files using Python, more specifically pickle.dump().
Sadly I get a warning message telling me that the array size does not match the binary file, even though the array I pickled has the exact same size as specified in the input file (1000 x 1000). I am using float32 precision. 
When looking at the output, the density is 0 everywhere, except for a 10x10 region on the top left, where some entries are infinite. The array I pickled in python is only made up of 0s and some entries which are 3e19.
What could go wrong here?
Best Regards,
Sebastian

**Comment by TomGoffrey** [comment]
I'd assume you'd want to output the data in double precision, as by default that's what EPOCH uses. If that doesn't work, could you post your python code? Also, are you certain `pickle.dump()` doesn't also output some metadata?
